invisible_cities/cities/detsim_simulate_electrons_test.py

```diff
@@ -0,0 +1,78 @@
+import os
+import numpy  as np
+import tables as tb
+import pandas as pd
+
+from pytest import fixture
+
+from invisible_cities.core.configure import configure
+
+from invisible_cities.io.mcinfo_io import read_mchits_df
+
+from invisible_cities.cities.detsim_simulate_electrons import *
+
+
+@fixture(params = ["Kr83_nexus_v5_03_00_ACTIVE_7bar_3evts.MCRD.h5"])
+def nexusMC_filename(request, ICDATADIR):
+    return os.path.join(ICDATADIR, request.param)
+
+
+@fixture
+def random_event_hits(nexusMC_filename):
+    filename = nexusMC_filename
+
+    with tb.open_file(filename) as h5in:
+        extents = pd.read_hdf(filename, 'MC/extents')
+        event_ids  = extents.evt_number
+        hits_df    = read_mchits_df(h5in, extents)
+
+        #choose random event
+        evt = np.random.choice(event_ids)
+        hits = hits_df.loc[evt, :, :]
+
+        print("Event", evt)
+
+        xs = hits["x"].values
+        ys = hits["y"].values
+        zs = hits["z"].values
+        energies = hits["energy"].values
+
+    return xs, ys, zs, energies
+
+
+@fixture
+def parameters():
+
+    conf = configure('detsim invisible_cities/config/detsim.conf'.split())
+
+    print(conf)
+
+    wi, fano_factor          = conf.get("wi")      , conf.get("fano_factor")
+    lifetime, drift_velocity = conf.get("lifetime"), conf.get("drift_velocity")
+    transversal_diffusion    = conf.get("transversal_diffusion")
+    longitudinal_diffusion   = conf.get("longitudinal_diffusion")
+
+    return wi, fano_factor, lifetime, drift_velocity,\
+           transversal_diffusion, longitudinal_diffusion
+
+
+def test_electron_simulation(random_event_hits, parameters):
+
+    xs, ys, zs, energies = random_event_hits
+
+    wi, fano_factor, lifetime, drift_velocity,\
+    transversal_diffusion, longitudinal_diffusion = parameters
+
+    # generate electrons
+    secondary_electrons = generate_electrons(energies, wi, fano_factor)
+    assert len(secondary_electrons)==len(energies)
+
+    # drift electrons
+    electrons = drift_electrons(zs, secondary_electrons, lifetime, drift_velocity)
+    assert np.sum(electrons)<=np.sum(secondary_electrons)
+
+    # diffuse electrons
+    (dxs, dys, dzs) = diffuse_electrons(xs, ys, zs, electrons,
+                                        transversal_diffusion,
+                                        longitudinal_diffusion)
+    assert len(dxs)==len(dys)==len(dzs)==np.sum(electrons)
```

Comment:
While it's quite nice to have a test which combines the three functions we need tests for each individual one too to make sure the functionality is there.

Also, we probably need to think of something a bit more robust than just checking the lengths of the outputs.